pkg/injector/webhook.go

```diff
@@ -180,20 +230,30 @@ func (wh *webhook) mutateHandler(w http.ResponseWriter, req *http.Request) {
 	log.Debug().Msg("Done responding to admission request")
 }
 
-func (wh *webhook) mutate(req *v1beta1.AdmissionRequest) *v1beta1.AdmissionResponse {
+func (wh *webhook) mutate(req *v1beta1.AdmissionRequest) (*v1beta1.AdmissionResponse, string) {
+	var outResourceName string
 	if req == nil {
 		log.Error().Msg("Nil AdmissionRequest")
-		return toAdmissionError(errors.New("nil admission request"))
+		return toAdmissionError(errors.New("nil admission request")), ""
 	}
 
 	// Decode the Pod spec from the request
 	var pod corev1.Pod
 	if err := json.Unmarshal(req.Object.Raw, &pod); err != nil {
 		log.Error().Err(err).Msg("Error unmarshaling request to Pod")
-		return toAdmissionError(err)
+		return toAdmissionError(err), ""
 	}
 	log.Info().Msgf("Mutation request: (new object: %v) (old object: %v)", string(req.Object.Raw), string(req.OldObject.Raw))
 
+	// Gather a meaningful resource name for this request
+	var pName string
+	if len(pod.Name) != 0 {
+		pName = pod.Name
+	} else {
+		pName = pod.GenerateName
+	}
+	outResourceName = fmt.Sprintf("%s/%s", req.Namespace, pName)
+
```

Comment:
Generating a pod name is not very meaningful. How would one compare this with actual k8s pods?